File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] ans = new int[nums.length];
        int l = 0, r = nums.length - 1, i = nums.length - 1;
        while(l <= r){
            if(Math.abs(nums[l]) > Math.abs(nums[r])){
                ans[i] = nums[l] * nums[l];
                l++;
            } else {
                ans[i] = nums[r] * nums[r];
                r--;
            }
            i--;
        }
        return ans;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = []
        for i in range(n):
            ans.append(0)
            nums[i] = nums[i]**2
        
        p = n - 1
        l = 0
        r = n - 1
        while l <= r:
            if nums[l] >= nums[r]:
                ans[p] = nums[l]
                l += 1
            else:
                ans[p] = nums[r]
                r -= 1
            p -= 1
        
        return ans